pkg/cache/memcached_client.go

```diff
@@ -181,7 +210,7 @@ func (c *memcachedClient) dialViaCircuitBreaker(network, address string, timeout
 	c.Unlock()
 
 	conn, err := cb.Execute(func() (interface{}, error) {
-		return net.DialTimeout(network, address, timeout)
+		return c.DialTimeout(network, address, timeout)
```

Comment:
it seems like this `DialTimeout` method is available here as `c.Client.DialTimeout` is there a reason to store it directly on the object?

Reply:
I imagine you could use `c.Client.DialTimeout` in lieu of storing it directly on the object. 

Personally, not so sure if there's any benefits of using one over the other here.

There doesn't seem to be any adverse effects from storing it directly on the object in Loki. Though, it could possibly differ in Tempo.